src/canvas/PathIntersection.tsx

```diff
@@ -25,7 +25,7 @@ export class PathIntersection {
         for (let i = 0; i < paths.length; i++) {
             const thatPath = paths[i];
 
-            const pointsOnThatPath = thatPath.getPoints();
+            const pointsOnThatPath = thatPath.getPoints().flat(1);
```

Comment:
The _findIntersectionOfLineSegmentAndPath uses the double array of points, because it needs to identify the index of the segment where the intersection happens. I guess we could use getSegments() there and compute points on each segment, but the whole point of keeping points in PatternPath was not to have to recompute all of them a every mouse move. So I see what you mean, but I couldn't see an alternative.